lib/siilar/client/radios.rb

```diff
@@ -98,10 +98,19 @@ def notify_favorite(radio, attributes = {})
       #
       # @see http://api.niland.io/2.0/doc/radios#notify-a-not-played-track
       def notify_not_played(radio, attributes = {})
-        Extra.validate_mandatory_attributes(attributes, [:track])
+        validate_arguments_has_track_or_reference(attributes)
         response = client.post("2.0/radios/#{radio}/notplayed", attributes)
         Struct::Radio.new(response)
       end
+
+      private
+ 
+      def validate_arguments_has_track_or_reference(attributes)
+        if !(attributes.key?(:track) ^ attributes.key?(:reference))
+          raise(ArgumentError, ":track or :reference is required. They are mutually exclusive")
+        end
```

Comment:
I wanted a XOR, because if both are provided in the same time it should also be false. Granted the XOR operator `^` is not very readable. 

Without the XOR I'm not sure which is the more elegant way to express this : 

```
if !(attributes[:track] || attributes[:reference]) || (attributes[:track] && attributes[:reference])
```